pkcs5/src/pbes2/encryption.rs

```diff
@@ -49,20 +55,44 @@ pub fn encrypt_in_place<'b>(
                 .encrypt(buffer, pos)
                 .map_err(|_| Error::EncryptFailed)
         }
+        #[cfg(feature = "ecb")]
+        EncryptionScheme::Aes128Ecb => {
+            let cipher = Aes128Ecb::new_from_slices(encryption_key.as_slice(), Default::default())
+                .map_err(|_| es.to_alg_params_invalid())?;
+            cipher
+                .encrypt(buffer, pos)
+                .map_err(|_| Error::EncryptFailed)
+        }
         EncryptionScheme::Aes192Cbc { iv } => {
             let cipher = Aes192Cbc::new_from_slices(encryption_key.as_slice(), iv)
                 .map_err(|_| es.to_alg_params_invalid())?;
             cipher
                 .encrypt(buffer, pos)
                 .map_err(|_| Error::EncryptFailed)
         }
+        #[cfg(feature = "ecb")]
+        EncryptionScheme::Aes192Ecb => {
+            let cipher = Aes192Ecb::new_from_slices(encryption_key.as_slice(), Default::default())
+                .map_err(|_| es.to_alg_params_invalid())?;
+            cipher
+                .encrypt(buffer, pos)
+                .map_err(|_| Error::EncryptFailed)
+        }
         EncryptionScheme::Aes256Cbc { iv } => {
             let cipher = Aes256Cbc::new_from_slices(encryption_key.as_slice(), iv)
                 .map_err(|_| es.to_alg_params_invalid())?;
             cipher
                 .encrypt(buffer, pos)
                 .map_err(|_| Error::EncryptFailed)
         }
+        #[cfg(feature = "ecb")]
+        EncryptionScheme::Aes256Ecb => {
+            let cipher = Aes256Ecb::new_from_slices(encryption_key.as_slice(), Default::default())
+                .map_err(|_| es.to_alg_params_invalid())?;
+            cipher
+                .encrypt(buffer, pos)
+                .map_err(|_| Error::EncryptFailed)
+        }
```

Comment:
Like 56-bit DES, I don't think it makes sense to support encrypting new files with known insecure modes. Instead, it should be decrypt-only.

Reply:
I'd missed you were doing that, I thought the feature-gating was sufficient discouragement.

Unfortunately for me I need to re-encrypt a response.
For me that feels somewhere in-between creating a new insecurely encrypted file and decrypting legacy content, but I can't think of a technical way to prevent making new files while still allowing participating in encrypted connections.

The application I'm working with takes other measures to protect against ECB-related insecurity.
Would an additional feature gate for insecure re-encryption be acceptable? I'm not sure how effective pushing for an encryption change and only supporting newer version of the app would be.

Reply:
I’d really prefer not to add encryption support for any known insecure modes

Reply:
Alright, I've taken encryption support out, up to you if you whether to merge or close since it won't have a known user.